src/torchcodec/_core/Encoder.cpp

```diff
@@ -92,14 +92,13 @@ AudioEncoder::AudioEncoder(
   validateSampleRate(*avCodec, sampleRate);
   avCodecContext_->sample_rate = sampleRate;
 
-  // Note: This is the format of the **input** waveform. This doesn't determine
-  // the output.
```

Comment:
My original comment was wrong: that's not the format of the input waveform. It's the format of the input AVFrame that we pass to `avcodec_send_frame()`. And it needs to be a format that the codec supports.